acceptance/indy_acceptance/indy-tests/test_scenarios/test_scenario_02.py

```diff
@@ -0,0 +1,157 @@
+import asyncio
+import json
+import logging
+import os
+import sys
+import time
+from indy import pool
+from indy.error import IndyError
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+from utils.constant import Constant, Colors
+from utils.utils import generate_random_string
+from utils.report import HTMLReport
+from utils.report import TestReport, Status
+
+
+# -----------------------------------------------------------------------------------------
+# This will run acceptance tests that will validate the add/remove roles functionality.
+# -----------------------------------------------------------------------------------------
+
+
+class MyVars:
+    # Data for generating report
+    test_name = "Test_Scenario_02_Verify_Messages_On_Connection"
+    test_report = TestReport(test_name)
+    the_error_message = "the information needed to connect was not found"
+    test_results = {"Step 1": False, "Step 2": False, "Step 3": False, "Step 4": False}
+
+    """  Needed some global variables. """
+    pool_handle = 0
+    pool_genesis_txn_file = Constant.pool_genesis_txn_file
+    original_pool_genesis_txn_file = Constant.original_pool_genesis_txn_file
+    pool_name = generate_random_string("test_pool", size=20)
+    debug = False
+
+    # cmds
+    back_up_pool_genesis_file = 'cp ' + pool_genesis_txn_file + " " + original_pool_genesis_txn_file
+    exit_sovrin = 'exit'
+    remove_pool_genesis_file = 'rm ' + pool_genesis_txn_file
+    restore_pool_genesis_file = 'cp ' + original_pool_genesis_txn_file + " " + pool_genesis_txn_file
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def run(cmd):
+    os.system(cmd)
+
+
+def test_precondition():
+    """  Make a copy of pool_transactions_sandbox_genesis  """
+    print(Colors.HEADER + "\nPrecondition \n" + Colors.ENDC)
+    run(MyVars.back_up_pool_genesis_file)
+    open(MyVars.pool_genesis_txn_file, 'w').close()
+
+
+async def test_scenario_02_verify_messages_on_connection():
+    logger.info("Test Scenario 02 -> started")
+
+    try:
+        # 1. Create ledger config from genesis txn file  ---------------------------------------------------------
+        step = "Step1.  Create Ledger"
+        print(Colors.HEADER + "\n\t {0}\n".format(step) + Colors.ENDC)
+        pool_config = json.dumps({"genesis_txn": str(MyVars.pool_genesis_txn_file)})
+
+        try:
```

Comment:
The same pattern/template is used for each test case (print/try/except).
Can we create an utility method and use this instead of boilerplate code (so that the whole try-except goes there, and we just pass a Callable with a test case)?
I think it will increase test readability and maintenance a lot.

Reply:
We created perform method in utils.py for print/try/except pattern.